go-controller/pkg/ovn/namespace.go

```diff
@@ -292,13 +292,20 @@ func (oc *Controller) updateNamespace(old, newer *kapi.Namespace) {
 				}
 				for _, pod := range existingPods {
 					logicalPort := util.GetLogicalPortName(pod.Namespace, pod.Name)
-					portInfo, err := oc.logicalPortCache.get(logicalPort)
+					if !util.PodWantsNetwork(pod) {
```

Comment:
since we are deleting something here I thought it was safe to try to delete it even if the pod is complete. We just want to avoid adding any new config for the pod.

Reply:
ack sg then! was just wondering if we'd hit the https://github.com/ovn-org/ovn-kubernetes/blob/bd4a607ca4ae304a624db858f90ad9c1c19db55d/go-controller/pkg/util/pod_annotation.go#L238 for the completed pods by any chance, but I think that warning could serve as a good indicator. Let's stick to your plan of better to try delete whenever possible.